README.md

```diff
@@ -75,6 +75,45 @@ api-addr: 127.0.0.1:8085
 data-dir: /data/bees/bee5
 ```
 
+## File upload
+
+File can be uploaded by making an HTTP request like this one:
+
+```sh
+curl -F file=@kitten.jpg http://localhost:8080/files
+```
+
+Curl will make a properly encoded `multipart/form-data` request sending the filename, content type and file content to the bee API, and it will return a response with the reference to the uploaded file:
+
+```json
+{"reference":"3b2791985f102fe645d1ebd7f51e522d277098fcd86526674755f762084b94ee"}
+```
+
+This reference is just an example, it will differ for every uploaded file.
+
+To download a file, open an URL with that reference `http://localhost:8080/files/3b2791985f102fe645d1ebd7f51e522d277098fcd86526674755f762084b94ee`.
+
+If you need to specify manually content type or filename during upload:
+
+```sh
+curl -H "Content-Type: image/x-jpeg" --data-binary @kitten.jpg localhost:8081/files?name=cat.jpg
+```
+
+The same response with file reference is returned.
```

Comment:
What is the purpose of having this here? Is it 1) a feature of the Bee node, or 2) just a different way to instruct curl to do the same? 

If 2, I think it would be better to omit it. If 1, it is not clear to me :)

Reply:
Curl is not doing the same. These are two different requests. The first one is multipart, the second one is streaming file content. This is the illustration what is possible and users can choose what suits better for their needs, as these examples also suggest what is possible if someone wants to use different tools or libraries, without going too technical on http.